src/libraries/Pool.sol

```diff
@@ -381,10 +381,17 @@ library Pool {
             // if the protocol fee is on, calculate how much is owed, decrement feeAmount, and increment protocolFee
             if (protocolFee > 0) {
                 unchecked {
-                    // step.amountIn does not include the swap fee, as it's already been taken from it,
-                    // so add it back to get the total amountIn and use that to calculate the amount of fees owed to the protocol
-                    // this line cannot overflow due to limits on the size of protocolFee and params.amountSpecified
-                    uint256 delta = (step.amountIn + step.feeAmount) * protocolFee / ProtocolFeeLibrary.PIPS_DENOMINATOR;
+                    uint256 delta;
+                    if (swapFee != protocolFee) {
```

Comment:
is this check equivalent to `if (lpFee != 0)` ?
I would find that more intuitive to read

Reply:
we cant access lpFee because its in a scoped block :(